lms/djangoapps/courseware/views/views.py

```diff
@@ -1544,7 +1544,7 @@ def financial_assistance_form(request):
             Q(_expiration_datetime__isnull=True) | Q(_expiration_datetime__gt=datetime.now(UTC())),
             course_id=enrollment.course_id,
             mode_slug=CourseMode.VERIFIED
-        ).exists()
+        ).exists() and enrollment.course_overview.eligible_for_financial_aid
```

Comment:
This has gotten too big to be a comprehension. I'd suggest moving it into a function. Also, you can do the `enrollment.course_overview.eligible_for_financial_aid` check before doing the filter to save a query.